cuqi/geometry/_geometry.py

```diff
@@ -409,6 +410,7 @@ def _plot_envelope(self, lo_values, up_values, **kwargs):
     def _plot_config(self):
         if self.axis_labels is not None:
             plt.xlabel(self.axis_labels[0])
+        plt.gca().xaxis.set_major_locator(MaxNLocator(integer=True))
```

Comment:
Since this belongs to Continous1D, I find it a bit strange to restrict tick locations to integers only.  Perhaps a more robust solution is to update the behavior of the default Geometry - which I believe is the cause of the described problem in the issue - by e.g. subclassing from a more generic class than Continous1D. Perhaps we can make a default Geometry class that has certain behavior when # of x-valeus are low, and another for higher.

Reply:
I agree here with @nabriis that setting the ticks to integer might not be the best for `Continuous1D`. One possibility is to override `_plot_config` in  `_DefaultGeometry` to have this line (line 413), and maybe set the display to marker (dots or o's) rather than lines. Or make the `_DefaultGeometry` a subclass of discrete instead of `Continuous1D`. I am not sure though what is the best solution.

You can make this another pull request if you like so the Samples __repr__ can be merged soon.

Reply:
Thank you both for your guidance and thoughts here. I now understand my update at Continuous1D didn't make sense at all and the existing methods to handle plot is already very good. I quote from @nabriis , "There is not great solution without being explicit and setting geometry manually". So, I'll withdraw my naive change here and leave the choice to the user.

Reply:
Ok let us leave it like it is then.